tests/ingest_runs/test_genomic_data_loader.py

```diff
@@ -0,0 +1,299 @@
+    df["Filename"] = df["Key"].map(lambda x: os.path.split(x)[-1])
+    mock = mocker.patch(
+        "creator.ingest_runs.genomic_data_loader.utils.scrape_s3",
+        return_value=df
+    )
+    return mock
+
+
+@pytest.fixture
+def mock_get_entities(mocker, study_generator):
+    """
+    Mock for creator.ingest_runs.utils.get_entities
+    """
+    def fake_get_entities(*args, **kwargs):
+        entity_type = args[1]
+        return study_generator.fake_entities[entity_type]
+
+    mock = mocker.patch(
+        "creator.ingest_runs.genomic_data_loader.utils.get_entities",
+        side_effect=fake_get_entities
+    )
+    return mock
+
+
+@pytest.fixture
+def mock_load_entities(tmpdir, mocker, study_generator):
+    """
+    Mock for GenomicDataLoader._load_entities function which allows testing to
+    occur without the use of the DataService.
+    """
+    def _dry_load_entities(*args, **kwargs):
+        """
+        Inserts KF IDs into output of _load_entities
+        """
+        from kf_lib_data_ingest.app.settings.base import TARGET_API_CONFIG
+        entity_type = args[0]
+        loader = LoadStage(
+            TARGET_API_CONFIG,
+            settings.DATASERVICE_URL,
+            [entity_type],
+            study_generator.study_id,
+            cache_dir=os.path.join(tmpdir, "temp"),
+            dry_run=True,
+        )
+        loader.run({entity_type: args[1]})
+        return loader.uid_cache
+
+    mock = mocker.patch(
+        "creator.ingest_runs.genomic_data_loader.GenomicDataLoader."
+        "_load_entities",
+        side_effect=_dry_load_entities
+    )
+    return mock
+
+
+@pytest.fixture
+def mock_load_bs_gf_links(mocker):
+    mock = mocker.patch(
+        "creator.ingest_runs.genomic_data_loader.GenomicDataLoader."
+        "load_specimen_harmonized_gf_links"
+    )
+    return mock
+
+
+@pytest.fixture
+def mock_load_se_gf_links(mocker):
+    mock = mocker.patch(
+        "creator.ingest_runs.genomic_data_loader.GenomicDataLoader."
+        "load_seq_exp_harmonized_genomic_files"
+    )
+    return mock
+
+
+@pytest.fixture
+def mock_validate(mocker):
+    mock = mocker.patch(
+        "creator.ingest_runs.genomic_data_loader.GenomicDataLoader."
+        "_validate_load"
+    )
+    return mock
+
+
+def test_load_harmonized_genomic_files(
+    mock_s3_scrape, study_generator, mock_load_entities
+):
+    """
+    Test the loading of the harmonized genomic files from the genomic workflow
+    output manifest.
+    """
+    loader = GenomicDataLoader(FAKE_STUDY)
+    manifest_df = study_generator.dataframes['gwo_manifest.tsv']
+    loader.load_harmonized_genomic_files(manifest_df)
+    mock_s3_scrape.assert_called_once()
+    mock_load_entities.assert_called_once()
+    entity_type, df = mock_load_entities.call_args_list[0][0]
+    assert isinstance(df, pd.DataFrame)
+    assert entity_type == GEN_FILE
+    assert not df.empty
+    manifest_df = manifest_df.drop_duplicates(["KF Participant ID"])
```

Comment:
Sorry I missed this before but...shouldn't this drop duplicates on something that uniquely identifies the harmonized genomic files? In this test we want to know that every harmonized gf in the manifest got submitted to the Dataservice. Dropping duplicates on participant ID would potentially get rid of rows you don't want to get rid of. The manifest might have rows with duplicate participant IDs if each participant has multiple specimens or if each participant has single specimen with multiple seq files per specimen. 

Actually thinking about it some more...do we even want to modify the manifest_df at all before comparing it to df2? Maybe we should just be comparing manifest_df rows to df2 rows. If they're equal then that means the inner join between the S3 scrape and the manifest didn't lose any rows

Reply:
These tests might all be working as is bc the test manifest_df might be inaccurate (my fault sorry 😬 ). If so, we should fix that.

Reply:
Yeah... This is a point that I've been confused about before.

Reply:
Here's a snippet of the manifest df with the s3 paths truncated for visual purposes:
<img width="1308" alt="image" src="https://user-images.githubusercontent.com/10283712/115793793-bdd3d680-a39a-11eb-84d2-2c2f11bd23a0.png">

This looks correct to me. We have a participant with one specimen that went through sequencing which produced 3 source/unharmonized files (1 cram, 1 crai, and 1 md5). The BIX team harmonized this and produced 1 harmonized file, the gVCF. The 3 rows are there because some information is repeated since there are 3 different source sequencing filles. So I think everything I said above still stands. You always want to do the "checking" with respect to the harmonized files since that's what we're loading into the Dataservice.

Reply:
Makes sense. So it seems that dropping duplicates on `Filepath` is what we want. I do think we need to drop rows for all these tests though. This manifest has 30 rows and 10 harmonized genomic files. So then we'd have 10 biospecimen-genomic-file links and 10 genomic-file-sequencing-experiment links as well.